### PR TITLE
chore(reth-node-builder): Replace HeadersClient + BodiesClient with BlockClient

### PR DESCRIPTION
**Pull request by SkandaBhat**
Replace HeadersClient + BodiesClient with BlockClient in reth-node-builder. https://github.com/paradigmxyz/reth/issues/10053

### CODE REVIEW

File: crates/node/builder/src/setup.rs [hunk 1]
```diff
@@ -39,7 +40,7 @@ pub fn build_networked_pipeline<DB, Client, Executor>(
 ) -> eyre::Result<Pipeline<DB>>
 where
     DB: Database + Unpin + Clone + 'static,
-    Client: HeadersClient + BodiesClient + Clone + 'static,
+    Client: BlockClient + Clone + 'static,
```
Comment (emhane) [inline crates/node/builder/src/setup.rs:43]:
```suggestion
    Client: BlockClient + 'static,
```
`Clone` is already supertrait of `BlockClient`

**Review by emhane — CHANGES_REQUESTED** [review]

File: crates/node/builder/src/setup.rs (hunk 1 above)
Comment (SkandaBhat) [reply crates/node/builder/src/setup.rs]: Resolved in https://github.com/paradigmxyz/reth/commit/a00d119bd93e09d554a167f40a5d3aaeb1a86a6c

**Review by emhane — APPROVED** [review]

**Comment by emhane** [comment]
recommend using the shortcut command `make lint` @SkandaBhat

**Comment by SkandaBhat** [comment]
> recommend using the shortcut command `make lint` @SkandaBhat

Thanks for the tip! And sorry for the multiple review requests, I am new to the repo and rust in general.

**Comment by emhane** [comment]
> Thanks for the tip! And sorry for the multiple review requests, I am new to the repo and rust in general.

no worries! thanks! hope you find some more issues that look fun to you

**Pull request merged**